sky/shell/platform/mac/platform_view_mac.mm

```diff
@@ -77,9 +145,15 @@
 void PlatformViewMac::RunFromSource(const std::string& main,
                                     const std::string& packages,
                                     const std::string& assets_directory) {
-  // TODO(johnmccutchan): Call to the Mac UI thread so that services work
-  // properly like we do in PlatformViewAndroid.
-  engine().RunFromSource(main, packages, assets_directory);
+  auto latch = new ftl::ManualResetWaitableEvent();
+
+  dispatch_async(dispatch_get_main_queue(), ^{
+    SetupAndLoadFromSource(main, packages, assets_directory);
+    latch->Signal();
+  });
+
+  latch->Wait();
+  delete latch;
```

Comment:
Same note about the Objective-C block capture of C++ objects. This will eventually be DRY'ed up across iOS/Mac platform views.